### PR TITLE
feature/audio-track-language-convenient-access

### PR DESCRIPTION
**Pull request by btmo**
Adds convenience methods to access the audio track language of Onix products and simplifies some methods.

Edit:
Added access to the default language of text as well.

### CODE REVIEW

File: lib/onix/descriptive_detail.rb [hunk 1]
```diff
@@ -516,12 +517,11 @@ def protections
     end
 
     def language_of_text
-      l=@languages.of_text.first
-      if l
-        l.code
-      else
-        nil
-      end
+      @languages.of_text.first&.code
```
Comment (fanaugen) [inline lib/onix/descriptive_detail.rb]: ❤️

File: lib/onix/onix21.rb [hunk 2]
```diff
@@ -459,6 +459,21 @@ def language_name_of_text
         end
       end
 
+      def language_of_audio_track
+        @languages
+          .select { |l| l.role.human == 'LanguageOfAudioTrack' }
```
Comment (fanaugen) [inline lib/onix/onix21.rb]: Pretty sure that you can save 1 line here using `.detect { ... }`, which is equivalent to `.select { ... }.first`

File: lib/onix/onix21.rb [hunk 3]
```diff
@@ -459,6 +459,21 @@ def language_name_of_text
...
+          .first
+          &.code
+      end
+
+      def language_code_of_audio_track
+        language_of_audio_track&.code
```
Comment (fanaugen) [inline lib/onix/onix21.rb]: doesn’t `language_of_audio_track` already return the language code?

Comment (btmo) [reply lib/onix/onix21.rb]: It returns an `Onix::LanguageCode` object which in turn has two methods `code` and `human` which then return actual strings.

File: lib/onix/onix21.rb (hunk 2 above)
Comment (btmo) [reply lib/onix/onix21.rb]: Done.

**Comment by btmo** [comment]
> `language_code_of_audio_track` and `language_of_audio_track` both seem to return the **code**

One returns a complex object `LanguageCode`, the other a string representation of the code.

**Pull request merged**